### ISSUE TITLE
Make IKnownType and KnowTypes public

### ISSUE THREAD

**Issue by MichaelRobl** [YAXLib/YAXLib#93]
Often we have a problem that we shoud serialize a class which is generic and we don't know the correct
type of a property at design time. One design constraint is that the type should be serializable. So the user of our class will save or load it and this will fail if the generic type is non serilaizable (in our case FileInfo for example). We could not apply the YaxCustomSerializer-Attribute on the generic property because at compile time the tpe is not defined.

here an dummy example;

```c#
public class Dummy<T>
{
 public DateTime LastAccess {get; set;}
 public T Source{get; set;}
}
```

example of using:

```c#
var d = new Dummy<FileInfo>();
var serializer = new YaxSerializer<Dummy<FileInfo>>();
serializer.SerializeToFile(d, "c:\\temp\\test.xml");
```

So after I have inspected the source code of YaxLib I think this problem could be solved if the interface IKnownType and the KnowTypes class will be made public. In this case we could derive from IKnownType  (FileInfoKnownType) and register this type in KnownTypes.

This will allow the users of our libary to register a static custom serializer  which could be used for all serialization-tasks!

**Comment by 304NotModified** [comment]
I guess the real feature request is to register custom Known types

**Comment by axunonb** [comment]
Resolved with release [v4.0.0](https://github.com/YAXLib/YAXLib/releases/tag/v4.0.0)
